File: model/restaurant.go
package model

import "gorm.io/gorm"

// Restaurant struct
type Restaurant struct {
	gorm.Model
	Name        string `gorm:"not null" json:"title"`
	Type        string `gorm:"not null" json:"type"`
	Location    int    `gorm:"not null" json:"location"`
	Phone       string `gorm:"not null" json:"phone"`
	Description string `gorm:"not null" json:"description"`
	Image       string `gorm:"not null" json:"image"`
}

File: web/src/app/model/restaurant.ts
export class RestauInfo {
  title: string;
  type: string;
  location: string;
  phone: string;
  description: string;
  image: string;

  constructor(title: string,
    type: string,
    location: string,
    phone: string,
    description: string,
    image: string) {
    this.title = title;
    this.type = type;
    this.location = location;
    this.phone = phone;
    this.description = description;
    this.image = image;
  }
}

export class RestauResponse {
  id: number;
  restauinfo: RestauInfo;
  favorited: boolean;

  constructor(id: number, restauinfo: RestauInfo, favorited: boolean) {
    this.id = id;
    this.restauinfo = restauinfo;
    this.favorited = favorited;
  }
}
